utils/ansible/ftest/templates/daos-launch.sh.j2

```diff
@@ -0,0 +1,192 @@
+		   -h, --help               show this help message and exit
+		   -V, --version            show version number
+		   -q, --quiet              quiet mode
+		   -v, --verbose            verbose mode
+		   -D, --debug              debug mode
+	EOF
+}
+
+function run
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_STANDARD ]]
+	then
+		"$@"
+	else
+		"$@" &> /dev/null
+	fi
+}
+
+{% raw %}
+opts=()
+args=()
+for item in "$@" ; do
+	if [[ ${#args} -eq 0 && "$item" != "--" ]] ; then
+		[[ ${item::1} == "-" ]] || fatal "unrecognized option \"$item\""
+		opts+=("$item")
+	else
+		args+=("$item")
+	fi
+done
+args=("${args[@]:1}")
+{% endraw %}
+
+OPTIONS=$(getopt -o "hVvDq" --long "help,version,verbose,debug,quiet" -- "${opts[@]}") || exit 1
+eval set -- "$OPTIONS"
+while true
+do
+	case "$1" in
+		-h|--help) usage ; exit 0;;
+		-V|--version) echo "daos-launch.sh version=$VERSION" ; exit 0 ;;
+		-v|--verbose) TRACE_LEVEL=$TRACE_LEVEL_VERBOSE ; shift 1 ;;
+		-D|--debug) TRACE_LEVEL=$TRACE_LEVEL_DEBUG ; set -x ; shift 1 ;;
+		-q|--quiet) TRACE_LEVEL=$TRACE_LEVEL_QUIET ; shift 1 ;;
+		--) shift ; break ;;
+		*) fatal "unrecognized option" ;;
+	esac
+done
+
+check_cmds $PYTHON_EXE $CLUSH_EXE $MODULE_EXE
+
+{% if "daos_clients" in groups and groups["daos_clients"] | length > 0 %}
+info "Clean up of clients $CLIENTS_LIST"
+$CLUSH_EXE -l root -w $CLIENTS_LIST sudo systemctl stop daos_agent
+$CLUSH_EXE -l root -w $CLIENTS_LIST rm -rf /var/run/daos_agent
+$CLUSH_EXE -l root -w $CLIENTS_LIST mkdir -p /var/run/daos_agent
+{% endif %}
+
+info "Clean up of servers $SERVERS_LIST"
+$CLUSH_EXE -l root -w $SERVERS_LIST sudo systemctl stop daos_server
+$CLUSH_EXE -l root -w $SERVERS_LIST rm -rf /var/run/daos_server
+$CLUSH_EXE -l root -w $SERVERS_LIST mkdir -p /var/run/daos_server
+
+info "Stopping all DAOS processes"
+{% if "daos_clients" in groups and groups["daos_clients"] | length > 0 %}
+CLUSH_NODESET="-w $SERVERS_LIST -w $CLIENTS_LIST"
+{% else %}
+CLUSH_NODESET="-w $SERVERS_LIST"
+{% endif %}
+$CLUSH_EXE -l root $CLUSH_NODESET 'bash -c "killall -q -9 avocado orterun mpirun orted daos_server daos_io_server daos_agent || true"'
+
+info "Setting simulation environment"
+export PYTHONPATH="$DAOS_INSTALL_DIR/lib64/python3.6/site-packages:$DAOS_INSTALL_DIR/lib/daos/TESTING/ftest/util"
+cd "$DAOS_SOURCE_DIR/utils/sl"
+source setup_local.sh build_vars-Linux.sh
+$MODULE_EXE load /etc/modulefiles/mpi/mlnx_openmpi-x86_64
+
+info "Parametrization of the simulation"
+cd "$DAOS_INSTALL_DIR/lib/daos/TESTING/ftest"
+{% if "daos_clients" in groups and groups["daos_clients"] | length > 0 %}
+common_opts="-card --test_servers=$SERVERS_LIST --test_clients=$CLIENTS_LIST"
+{% else %}
+common_opts="-card --test_servers=$SERVERS_LIST"
+{% endif %}
+
+info "Starting functional test simulation"
+{% if daos_ofi_interface is defined %}
+env OFI_INTERFACE=$OFI_INTERFACE $PYTHON_EXE launch.py $common_opts "${args[@]}"
+{% else %}
+$PYTHON_EXE launch.py $common_opts "${args[@]}"
+{% endif %}
```

Comment:
One thing that could be useful is the `umount` and `wipefs` cleanup steps after running the tests. If you run Avocado back to back, it's not necessary, but if you were to run the server manually after running Avocado, you most likely need to do these cleanup steps, which I find it bothersome.

Reply:
- [ ] Add option to cleanup after simulation